src/resolver/generics.rs

```diff
@@ -335,25 +340,30 @@ impl<'i> TypeAnnotator<'i> {
         generic_name_resolver: GenericNameResolver,
     ) -> (String, StatementAnnotation) {
         let call_name = generic_name_resolver(qualified_name, generics, generic_map);
-        let return_type = if let DataTypeInformation::Generic { generic_symbol, .. } =
-            self.index.get_type_information_or_void(return_type)
-        {
-            generic_map
-                .get(generic_symbol)
-                .map(String::as_str)
-                .unwrap_or(return_type)
-        } else {
-            return_type
-        }
-        .to_string();
-        (
-            call_name.clone(),
-            StatementAnnotation::Function {
-                qualified_name: qualified_name.to_string(),
-                return_type,
-                call_name: Some(call_name),
-            },
-        )
+        let annotation = self
```

Comment:
lets rename and document this function
better name: get_specific_function_annotation and in the parameters ask for: generic_qualified_name &generic_return_type

takes the generic signature of a function generic function and resolves it to its specific types according to the generic_map and the generic_name_resolver.